src/blockchain/blockchain.service.ts

```diff
@@ -165,60 +167,75 @@ export class BlockchainService implements OnApplicationBootstrap, OnApplicationS
     return (await this.apiPromise.rpc.chain.getFinalizedHead()) as BlockHash;
   }
 
+  private checkTotalCapacityLimit(capacityInfo: { remainingCapacity: bigint; totalCapacityIssued: bigint }, totalLimit: ICapacityLimit): boolean {
+    const { remainingCapacity, totalCapacityIssued } = capacityInfo;
+    const totalCapacityUsed = totalCapacityIssued - remainingCapacity;
+    let outOfCapacity = false;
+
+    if (totalLimit.type === 'percentage') {
+      const percentLimit = (totalCapacityIssued * totalLimit.value) / 100n;
+      outOfCapacity = totalCapacityUsed >= percentLimit;
+    } else if (totalLimit.type === 'amount') {
+      outOfCapacity = totalCapacityUsed >= totalLimit.value;
+    }
+
+    if (outOfCapacity) {
+      this.logger.warn(`Total capacity usage limit reached: used ${totalCapacityUsed} of ${totalCapacityIssued}`);
+    }
+    return outOfCapacity;
+  }
+
+  private async checkServiceCapacityLimit(
+    capacityInfo: { remainingCapacity: bigint; totalCapacityIssued: bigint; currentEpoch: bigint },
+    serviceLimit: ICapacityLimit,
+  ): Promise<boolean> {
+    const { remainingCapacity, totalCapacityIssued, currentEpoch } = capacityInfo;
+    let limit: bigint;
+    if (serviceLimit.type === 'percentage') {
+      limit = (totalCapacityIssued * serviceLimit.value) / 100n;
+    } else if (serviceLimit.type === 'amount') {
+      limit = serviceLimit.value;
+    } else {
+      throw new Error('Unknown capacity limit');
+    }
+
```

Comment:
this can move to bootstrap? or it will be changed often via config changes?

Reply:
The percentage won't change, but the amount of capacity it represents will change as provider total amount staked/total capacity issued changes.